In [1]:
import pandas as pd
import numpy as np
import seaborn as sns
import matplotlib.pyplot as plt
import datetime
import matplotlib.dates as md
import os

%matplotlib inline

In [4]:
df = pd.read_csv(r"C:\Users\안지은\Desktop\Team.Cayley\owid-covid-data (1).csv")
df.head()

iso_code continent     location        date  total_cases  new_cases  \
0      AFG      Asia  Afghanistan  2020-02-24          1.0        1.0   
1      AFG      Asia  Afghanistan  2020-02-25          1.0        0.0   
2      AFG      Asia  Afghanistan  2020-02-26          1.0        0.0   
3      AFG      Asia  Afghanistan  2020-02-27          1.0        0.0   
4      AFG      Asia  Afghanistan  2020-02-28          1.0        0.0   

   new_cases_smoothed  total_deaths  new_deaths  new_deaths_smoothed  ...  \
0                 NaN           NaN         NaN                  NaN  ...   
1                 NaN           NaN         NaN                  NaN  ...   
2                 NaN           NaN         NaN                  NaN  ...   
3                 NaN           NaN         NaN                  NaN  ...   
4                 NaN           NaN         NaN                  NaN  ...   

   gdp_per_capita  extreme_poverty  cardiovasc_death_rate  \
0        1803.987              NaN                597.029   
1        1803.987              NaN                597.029   
2        1803.987              NaN                597.029   
3        1803.987              NaN                597.029   
4        1803.987              NaN                597.029   

   diabetes_prevalence  female_smokers  male_smokers  handwashing_facilities  \
0                 9.59             NaN           NaN                  37.746   
1                 9.59             NaN           NaN                  37.746   
2                 9.59             NaN           NaN                  37.746   
3                 9.59             NaN           NaN                  37.746   
4                 9.59             NaN           NaN                  37.746   

   hospital_beds_per_thousand  life_expectancy  human_development_index  
0                         0.5            64.83                    0.498  
1                         0.5            64.83                    0.498  
2                         0.5            64.83                    0.498  
3                         0.5            64.83                    0.498  
4                         0.5            64.83                    0.498  

[5 rows x 54 columns]

In [71]:
df.drop(df[df.location == 'Hong Kong'].index, 0,inplace = True)

In [77]:
df = df.reset_index()

In [79]:
df.drop('index', 1, inplace = True)
df

iso_code continent     location        date  total_cases  new_cases  \
0          AFG      Asia  Afghanistan  2020-02-24          1.0        1.0   
1          AFG      Asia  Afghanistan  2020-02-25          1.0        0.0   
2          AFG      Asia  Afghanistan  2020-02-26          1.0        0.0   
3          AFG      Asia  Afghanistan  2020-02-27          1.0        0.0   
4          AFG      Asia  Afghanistan  2020-02-28          1.0        0.0   
...        ...       ...          ...         ...          ...        ...   
58872      ZWE    Africa     Zimbabwe  2021-01-04      15829.0      564.0   
58873      ZWE    Africa     Zimbabwe  2021-01-05      17194.0     1365.0   
58874      ZWE    Africa     Zimbabwe  2021-01-06      17804.0      610.0   
58875      ZWE    Africa     Zimbabwe  2021-01-07      18675.0      871.0   
58876      ZWE    Africa     Zimbabwe  2021-01-08      19660.0      985.0   

       new_cases_smoothed  total_deaths  new_deaths  new_deaths_smoothed  ...  \
0                     NaN           NaN         NaN                  NaN  ...   
1                     NaN           NaN         NaN                  NaN  ...   
2                     NaN           NaN         NaN                  NaN  ...   
3                     NaN           NaN         NaN                  NaN  ...   
4                     NaN           NaN         NaN                  NaN  ...   
...                   ...           ...         ...                  ...  ...   
58872             383.000         384.0         4.0                4.286  ...   
58873             552.714         418.0        34.0                8.429  ...   
58874             597.000         431.0        13.0               10.143  ...   
58875             686.857         446.0        15.0               11.857  ...   
58876             796.571         468.0        22.0               14.143  ...   

       gdp_per_capita  extreme_poverty  cardiovasc_death_rate  \
0            1803.987              NaN                597.029   
1            1803.987              NaN                597.029   
2            1803.987              NaN                597.029   
3            1803.987              NaN                597.029   
4            1803.987              NaN                597.029   
...               ...              ...                    ...   
58872        1899.775             21.4                307.846   
58873        1899.775             21.4                307.846   
58874        1899.775             21.4                307.846   
58875        1899.775             21.4                307.846   
58876        1899.775             21.4                307.846   

       diabetes_prevalence  female_smokers  male_smokers  \
0                     9.59             NaN           NaN   
1                     9.59             NaN           NaN   
2                     9.59             NaN           NaN   
3                     9.59             NaN           NaN   
4                     9.59             NaN           NaN   
...                    ...             ...           ...   
58872                 1.82             1.6          30.7   
58873                 1.82             1.6          30.7   
58874                 1.82             1.6          30.7   
58875                 1.82             1.6          30.7   
58876                 1.82             1.6          30.7   

       handwashing_facilities  hospital_beds_per_thousand  life_expectancy  \
0                      37.746                         0.5            64.83   
1                      37.746                         0.5            64.83   
2                      37.746                         0.5            64.83   
3                      37.746                         0.5            64.83   
4                      37.746                         0.5            64.83   
...                       ...                         ...              ...   
58872                  36.791                         1.7   

In [163]:
df_loc_list = df.location.unique().tolist()

start_date = []

for n in range(len(df_loc_list)):
    start_rec_date = df.date[df.location == df_loc_list[n]][df.total_cases > 0 ].head(1).unique().tolist()
    start_rec_date_case = df.total_cases[df.location == df_loc_list[n]][df.total_cases > 0 ].head(1).unique().tolist()
    start_date.append([df_loc_list[n], start_rec_date,start_rec_date_case ])

# 편의를 위해 abc순 10개만 출력
start_date[:10]

[['Afghanistan', ['2020-02-24'], [1.0]],
 ['Albania', ['2020-03-09'], [2.0]],
 ['Algeria', ['2020-02-25'], [1.0]],
 ['Andorra', ['2020-03-02'], [1.0]],
 ['Angola', ['2020-03-20'], [1.0]],
 ['Antigua and Barbuda', ['2020-03-13'], [1.0]],
 ['Argentina', ['2020-03-03'], [1.0]],
 ['Armenia', ['2020-03-01'], [1.0]],
 ['Australia', ['2020-01-26'], [4.0]],
 ['Austria', ['2020-02-25'], [2.0]]]

In [169]:
start_date_loc = list()

for n in range(len(start_date)):
    start_date_loc.append(start_date[n][0])
start_date_loc

['Afghanistan',
 'Albania',
 'Algeria',
 'Andorra',
 'Angola',
 'Antigua and Barbuda',
 'Argentina',
 'Armenia',
 'Australia',
 'Austria',
 'Azerbaijan',
 'Bahamas',
 'Bahrain',
 'Bangladesh',
 'Barbados',
 'Belarus',
 'Belgium',
 'Belize',
 'Benin',
 'Bhutan',
 'Bolivia',
 'Bosnia and Herzegovina',
 'Botswana',
 'Brazil',
 'Brunei',
 'Bulgaria',
 'Burkina Faso',
 'Burundi',
 'Cambodia',
 'Cameroon',
 'Canada',
 'Cape Verde',
 'Central African Republic',
 'Chad',
 'Chile',
 'China',
 'Colombia',
 'Comoros',
 'Congo',
 'Costa Rica',
 "Cote d'Ivoire",
 'Croatia',
 'Cuba',
 'Cyprus',
 'Czechia',
 'Democratic Republic of Congo',
 'Denmark',
 'Djibouti',
 'Dominica',
 'Dominican Republic',
 'Ecuador',
 'Egypt',
 'El Salvador',
 'Equatorial Guinea',
 'Eritrea',
 'Estonia',
 'Eswatini',
 'Ethiopia',
 'Fiji',
 'Finland',
 'France',
 'Gabon',
 'Gambia',
 'Georgia',
 'Germany',
 'Ghana',
 'Greece',
 'Grenada',
 'Guatemala',
 'Guinea',
 'Guinea-Bissau',
 'Guyana',
 'Haiti',
 'Honduras',
 'Hungary

In [170]:
start_date_date = list()
for n in range(len(start_date)):
    start_date_date.append(start_date[n][1][0])
start_date_date

['2020-02-24',
 '2020-03-09',
 '2020-02-25',
 '2020-03-02',
 '2020-03-20',
 '2020-03-13',
 '2020-03-03',
 '2020-03-01',
 '2020-01-26',
 '2020-02-25',
 '2020-03-01',
 '2020-03-16',
 '2020-02-24',
 '2020-03-08',
 '2020-03-17',
 '2020-02-28',
 '2020-02-04',
 '2020-03-23',
 '2020-03-16',
 '2020-03-06',
 '2020-03-11',
 '2020-03-05',
 '2020-03-30',
 '2020-02-26',
 '2020-03-09',
 '2020-03-08',
 '2020-03-10',
 '2020-03-31',
 '2020-01-27',
 '2020-03-06',
 '2020-01-26',
 '2020-03-20',
 '2020-03-15',
 '2020-03-19',
 '2020-02-23',
 '2020-01-22',
 '2020-03-06',
 '2020-04-30',
 '2020-03-15',
 '2020-03-06',
 '2020-03-11',
 '2020-02-25',
 '2020-03-12',
 '2020-03-09',
 '2020-03-01',
 '2020-03-11',
 '2020-02-27',
 '2020-03-18',
 '2020-03-22',
 '2020-03-01',
 '2020-03-01',
 '2020-02-14',
 '2020-03-19',
 '2020-03-15',
 '2020-03-21',
 '2020-02-27',
 '2020-03-14',
 '2020-03-13',
 '2020-03-19',
 '2020-01-29',
 '2020-01-24',
 '2020-03-14',
 '2020-03-17',
 '2020-02-26',
 '2020-01-27',
 '2020-03-14',
 '2020-02-

In [173]:
names = start_date_loc
dates = [datetime.strptime(ii, "%Y-%m-%d") for ii in start_date_date]
dates

In [152]:
!pip install plotly

  Created wheel for retrying: filename=retrying-1.3.3-py3-none-any.whl size=11435 sha256=e053e349fca3d2d65204caf811f7d8840f59de9cf3cad6875763e351904800d8
  Stored in directory: c:\users\안지은\appdata\local\pip\cache\wheels\f9\8d\8d\f6af3f7f9eea3553bc2fe6d53e4b287dad18b06a861ac56ddf
Successfully built retrying


In [157]:
import plotly as py
from plotly.offline import iplot
import plotly.graph_objects as go
import numpy as np

In [168]:
start_date_case_list = list()
for n in range(len(start_date)):
    start_date_case_list.append(start_date[n][2][0])
start_date_case_list

[1.0,
 2.0,
 1.0,
 1.0,
 1.0,
 1.0,
 1.0,
 1.0,
 4.0,
 2.0,
 3.0,
 1.0,
 1.0,
 3.0,
 2.0,
 1.0,
 1.0,
 1.0,
 1.0,
 1.0,
 2.0,
 2.0,
 3.0,
 1.0,
 1.0,
 4.0,
 1.0,
 2.0,
 1.0,
 1.0,
 1.0,
 1.0,
 1.0,
 1.0,
 2.0,
 548.0,
 1.0,
 1.0,
 1.0,
 1.0,
 1.0,
 1.0,
 3.0,
 2.0,
 3.0,
 1.0,
 1.0,
 1.0,
 1.0,
 1.0,
 6.0,
 1.0,
 1.0,
 1.0,
 1.0,
 1.0,
 1.0,
 1.0,
 1.0,
 1.0,
 2.0,
 1.0,
 1.0,
 1.0,
 1.0,
 3.0,
 1.0,
 1.0,
 1.0,
 1.0,
 2.0,
 1.0,
 2.0,
 2.0,
 2.0,
 1.0,
 1.0,
 2.0,
 61.0,
 2.0,
 1.0,
 1.0,
 1.0,
 2.0,
 1.0,
 2.0,
 1.0,
 4.0,
 1.0,
 2.0,
 1.0,
 3.0,
 2.0,
 1.0,
 1.0,
 1.0,
 1.0,
 1.0,
 1.0,
 1.0,
 1.0,
 3.0,
 3.0,
 3.0,
 4.0,
 2.0,
 3.0,
 2.0,
 1.0,
 3.0,
 1.0,
 1.0,
 1.0,
 1.0,
 2.0,
 1.0,
 1.0,
 8.0,
 2.0,
 1.0,
 1.0,
 1.0,
 1.0,
 1.0,
 1.0,
 1.0,
 1.0,
 2.0,
 2.0,
 4.0,
 1.0,
 1.0,
 1.0,
 1.0,
 1.0,
 1.0,
 2.0,
 1.0,
 1.0,
 2.0,
 1.0,
 2.0,
 1.0,
 1.0,
 1.0,
 1.0,
 4.0,
 1.0,
 1.0,
 1.0,
 2.0,
 1.0,
 1.0,
 1.0,
 2.0,
 2.0,
 1.0,
 1.0,
 1.0,
 1.0,
 1.0,
 1.0,
 1.0,
 1.0,
 1.0,
 1.0,
 

In [210]:
df_start = pd.DataFrame(start_date_loc)

In [211]:
df_start['date'] = dates
df_start['case'] = start_date_case_list
df_start

0       date   case
0    Afghanistan 2020-02-24    1.0
1        Albania 2020-03-09    2.0
2        Algeria 2020-02-25    1.0
3        Andorra 2020-03-02    1.0
4         Angola 2020-03-20    1.0
..           ...        ...    ...
186      Vietnam 2020-01-23    2.0
187        World 2020-01-22  555.0
188        Yemen 2020-04-10    1.0
189       Zambia 2020-03-18    2.0
190     Zimbabwe 2020-03-20    1.0

[191 rows x 3 columns]

In [212]:
df_start['location'] = df_start[0]
df_start.drop(0,1, inplace = True)
df_start

date   case     location
0   2020-02-24    1.0  Afghanistan
1   2020-03-09    2.0      Albania
2   2020-02-25    1.0      Algeria
3   2020-03-02    1.0      Andorra
4   2020-03-20    1.0       Angola
..         ...    ...          ...
186 2020-01-23    2.0      Vietnam
187 2020-01-22  555.0        World
188 2020-04-10    1.0        Yemen
189 2020-03-18    2.0       Zambia
190 2020-03-20    1.0     Zimbabwe

[191 rows x 3 columns]

In [204]:
import plotly.express as px

In [237]:
fig = px.scatter(data_frame=df_start, x='date', y='case', hover_name='location' )
fig.show()

<h3>개선점</h3>

> <h4> 보기 불편한 감이 있음. 남는 시간에 더 수정할 수 있으면 검색하거나 특정 지역만 다른 색으로 표시할 수 있도록 할 것임.</h4> 

-------------------------------------------------

<h3> 후기 </h3>

> <h4> 주로 3-4월에 첫 확진자가 발생하기 시작했음을 알 수 있음.</h4> 
>
> <h4> 중국의 경우만 555명으로 값이 튀게 나오는데, 중국을 제외하거나 중국의 확진자 수를 1로 조정하여 값이 튀지 않게 해야할 지 고민중임.</h4> 
>
> <h4> international이 시작된 2월 7일 기사들을 보니, international이 다이아몬드 프린세스 호를 의미하는 것이 맞는 것 같음.</h4>


-------------------------------------------------
* 코드 참조 : https://hogni.tistory.com/87
* plotly : https://data101.oopy.io/plolty-tutorial-guide-in-korean

In [234]:
#추가) 시작일 찾아주는 함수
def find_start_record(location):
    start_record_date = df_start.date[df_start.location == location]
    return start_record_date

In [235]:
location = input()
find_start_record(location)

Uganda


176   2020-03-21
Name: date, dtype: datetime64[ns]